src/demos/linking-it-all/app.js

```diff
@@ -161,43 +134,46 @@ export default class App extends Component {
     });
   }
 
+  _updateLayerSettings(settings) {
+    this.setState({settings});
+  }
+
   render() {
-    const {viewport, hoveredObject, 
-      points, settings, status, x, y,
-      highlightedHour, selectedHour
-    } = this.state;
     return (
       <div>
-        {hoveredObject &&
-          <div style={{...tooltipStyle, left: x, top: y}}>
-            <div>{hoveredObject.id}</div>
+        {this.state.hoveredObject &&
+          <div style={{
+            ...tooltipStyle,
+            left: this.state.x,
+            top: this.state.y
+          }}>
+            <div>{this.state.hoveredObject.id}</div>
           </div>}
-        <LayerControls
-          settings={settings}
-          propTypes={LAYER_CONTROLS}
-          onChange={this.updateLayerSettings.bind(this)}
-        />
+        {this.props.noControls ? null : <LayerControls
+          settings={this.state.settings}
+          propTypes={HEXAGON_CONTROLS}
+          onChange={settings => this._updateLayerSettings(settings)}
+        />}
         <MapGL
-          {...viewport}
+          {...this.state.viewport}
           mapStyle={MAPBOX_STYLE}
-          onViewportChange={this._onViewportChange.bind(this)}
+          onViewportChange={viewport => {
+            this._onViewportChange(viewport);
+          }}
           mapboxApiAccessToken={MAPBOX_TOKEN}>
           <DeckGLOverlay
-            viewport={viewport}
-            data={points}
-            hour={highlightedHour || selectedHour}
-            onHover={this._onHover.bind(this)}
-            settings={settings}/>
+            viewport={this.state.viewport}
+            data={this.state.points}
+            hour={this.state.highlightedHour || this.state.selectedHour}
+            onHover={hover => this._onHover(hover)}
+            {...this.state.settings}
+          />
         </MapGL>
         <Charts {...this.state}
```

Comment:
We should hide this on the `Hero` too, it feels out of place. Or maybe move it to the right?

Reply:
we could. but it's so we have something to interact with. we could also pass a prop.